### PR TITLE
New version of cardOpt

### PR DESCRIPTION
**Pull request by laurentheirendt**
**Summary**

- adding the new version of cardOpt
- upgrading the version of `mve 1.0` -> `mve 1.1` (external)
- minor changes in `reconstruction`
- adding `testRemoveDuplicateRxn.m`
- removing `.tab` files and legacy conflict files

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [contributing](https://github.com/opencobra/cobratoolbox/blob/documentation/.github/CONTRIBUTING.md) document
- [X] Followed the [code styleguide](https://github.com/opencobra/cobratoolbox/blob/documentation/.github/CONTRIBUTING.md#code-styleguide)
- [X] Checked to ensure there aren't other open [Pull Requests](https://github.com/opencobra/cobratoolbox/pulls) for the same update/change
- [X] Written platform-independent code
- [X] Used `pwd` to get the current directory
- [X] Used `filesep` for paths (e.g., `['myPath' filesep 'myFile.m']`)
- [X] Made sure that computational results are reproducible
- [X] Documented your code based on the [Documentation styleguide](#documentation-and-comments-styleguide).
- [X] Documented the `Input` and `Output` arguments
- [X] Followed the guidelines to automatically generate the documentation

###  Tests tailored to test the PR

**I hereby confirm that I have:**

- [X] Written a sensible test according to the [Test styleguide](#test-styleguide)
- [X] Tested your PR locally prior to submission
- [X] Checked that all tests pass
- [X] Tested the code on Linux

### CODE REVIEW

File: src/checkDuplicateRxn.m [hunk 1]
```diff
@@ -1,78 +1,187 @@
...
-                nRxns = nRxns - length(matches);
-                model2 = removeRxns(model2,model.rxns(i));
-                model2.rxns{i} = model.rxns{i};
-                model = model2;
-                removed(cnt,1) = i;
-                cnt = cnt+1;
+        if printLevel>0
+            fprintf('%s\n','Checking for reaction duplicates by stoichometry ...');
+        end
+        if 0
```
Comment (laurentheirendt) [inline src/checkDuplicateRxn.m]: Fixed.

File: src/checkDuplicateRxn.m [hunk 2]
```diff
@@ -1,78 +1,187 @@
...
-			rxnRelationship.duplicates = duplicateReactions;
-		end
-		close(h);
-end
\ No newline at end of file
+        if printLevel>0
+            fprintf('%s\n','Checking for reaction duplicates by stoichometry (up to orientation) ...');
+        end
+        if 0
+            [~, ia, ic] = unique(model.S', 'rows');
```
Comment (laurentheirendt) [inline src/checkDuplicateRxn.m]: Fixed.

File: src/reconstruction/addReaction.m [hunk 3]
```diff
@@ -54,6 +54,10 @@
 if ~exist('addRxnGeneMat','var')
```
Comment (laurentheirendt) [inline src/reconstruction/addReaction.m]: Fixed.

File: src/cardOpt/stoichConsistency/findMassLeaksAndSiphons.m [hunk 4]
```diff
@@ -199,38 +225,37 @@
...
         cardPrb.c       = zeros(cardPrb.p+cardPrb.q+cardPrb.r,1);
         cardPrb.lambda  = 0;
         cardPrb.delta   = 1;
         cardPrb.A       = [-speye(mlt) S];
         cardPrb.b       = zeros(mlt,1);
         cardPrb.csense  = repmat('E',mlt, 1);
-                
-        cardPrb.lb      = [zeros(mlt,1)-feasTol;lb];%helps keep the problem feasible
+        cardPrb.lb      = [zeros(mlt,1);lb];             
+        %cardPrb.lb      = [zeros(mlt,1)-feasTol;lb];%helps keep the problem feasible
```
Comment (laurentheirendt) [inline src/cardOpt/stoichConsistency/findMassLeaksAndSiphons.m]: Fixed.

File: src/cardOpt/stoichConsistency/findStoichConsistentSubset.m [hunk 5]
```diff
@@ -63,72 +110,105 @@
...
-        %mass and charge balance can be checked by looking at formulas
-        if isfield(model,'metFormulas')
-            [massImbalance,imBalancedMass,imBalancedCharge,imBalancedRxnBool,Elements,missingFormulaeBool,balancedMetBool]...
-                = checkMassChargeBalance(model,printLevel-1);
-            model.balancedRxnBool=~imBalancedRxnBool;
-            model.balancedMetBool=balancedMetBool;
-            
-            model.Elements=Elements;
-            model.missingFormulaeBool=missingFormulaeBool;
+        if 1
```
Comment (laurentheirendt) [inline src/cardOpt/stoichConsistency/findStoichConsistentSubset.m]: Fixed.

File: src/cardOpt/stoichConsistency/findStoichConsistentSubset.m [hunk 6]
```diff
@@ -63,72 +110,105 @@
...
+    
+    %% minimum cardinality of conservation relaxation vector
+    [relaxRxnBool,solutionRelax] = minCardinalityConservationRelaxationVector(model.S(:,model.balancedRxnBool),minCardRelaxParams,printLevel-1);
+    minConservationNonRelaxRxnBool=false(nRxn,1);
+    minConservationNonRelaxRxnBool(model.balancedRxnBool)=~relaxRxnBool;
+    minConservationNonRelaxMetBool = getCorrespondingRows(model.S,true(nMet,1),minConservationNonRelaxRxnBool,'exclusive');
+    
+    %keeps the mass balanced reactions as part of the non-relaxed reaction
+    %set when testing for the minimal number of relaxed reactions
+    if nnz(model.balancedRxnBool)==nnz(minConservationNonRelaxRxnBool) && 0
```
Comment (laurentheirendt) [inline src/cardOpt/stoichConsistency/findStoichConsistentSubset.m]: Fixed.

File: src/cardOpt/stoichConsistency/findStoichConsistentSubset.m [hunk 7]
```diff
@@ -63,72 +110,105 @@
...
-%assumes that all mass imbalanced reactions are exchange reactions
-model.SIntMetBool = model.SIntMetBool & model.balancedMetBool;
-model.SIntRxnBool = model.SIntRxnBool & model.balancedRxnBool;
-
-if massBalanceCheck
-    if printLevel>0
-        fprintf('%6u\t%6u\t%s\n',nnz(model.SIntMetBool),nnz(model.SIntRxnBool),' heuristically non-exchange and seemingly elementally balanced.')
-        fprintf('%6u\t%6u\t%s\n',nnz(~model.SIntMetBool),nnz(~model.SIntRxnBool),' heuristically exchange and seemingly elementally imbalanced.')
-    end
+if 0
```
Comment (laurentheirendt) [inline src/cardOpt/stoichConsistency/findStoichConsistentSubset.m]: Fixed.

**Comment by rmtfleming** [comment]
Hi Thomas, we were working on this at the same time. Please can you manually merge these versions with both our added functionalities in the end result?

**Pull request merged**